pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -48,12 +49,19 @@ type MasterConfig struct {
 }
 
 func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextMapper kapi.RequestContextMapper, kubeClient *kclient.Client, projectCache *projectcache.ProjectCache) (*MasterConfig, error) {
+	var etcdClient *etcdclient.Client
+	var err error
+
 	if options.KubernetesMasterConfig == nil {
 		return nil, errors.New("insufficient information to build KubernetesMasterConfig")
 	}
 
 	// Connect and setup etcd interfaces
-	etcdClient, err := etcd.EtcdClient(options.EtcdClientInfo)
+	if options.KubernetesEtcdClientInfo != nil {
```

Comment:
I would rather populate KubernetesEtcdClientInfo from EtcdClientInfo in defaulting than make all the code have to check double conditions.